lib/apickli.js

```diff
@@ -198,6 +198,30 @@ const _assertResponseBodyPathIsArray = curry((path, pair) =>
   )
 )
 
+const _assertResponseBodyPathIsArrayWithLength = curry((path, len, pair) =>
+  pair.merge((ctx, res) =>
+    util.evaluatePath(path, res.body)
+    .either(
+      e => assert.fail(`Error during path evaluation: ${e}`),
+      result =>
+        result.either(
+          () => assert.fail(`Path evaluation did not find any matches for ${path}`),
+          x => {
+            assert.isTrue(
+              Array.isArray(x),
+              'Response body path is array'
+            )
+            assert.equal(
+              len,
+              x.length,
+              'Response body path is the expected length'
+            )
+          
+        })
+    )
+  )
+)
```

Comment:
I don't like the way I have implemented this. I would like this function to reuse the logic in `_assertResponseBodyPathIsArray` rather than duplicated it. However I am currently experimenting with the following things:

- Should the sequence of assertBodyPathIsArray and assertArrayLength be defined in apickli-gherkin.js or apickli.js
- Should I just create a vanilla assertArrayLength in Utils? Maybe this is a bad idea as currently all the assertions are in apickli-gherkin.js
- Should I use Lenses or create a getObjectAtPath to do something like this in Apickli.js...

```js
Then(/^response body path (.+) should be of type array with length (\d*)$/, function(path, cb) {
  Task.of(this.responsePair)
  .chain(apickli.assertResponseBodyPathIsArrayWithLength(path))
  .chain(util.assertArrayLength(apickli.getObjectAtPath(path)))    //no idea how this would actually work, just thinking of the structure
  .fork(
    e => cb(e),
    () => cb()
  )
})
```

This means I could reuse assertResponseBodyPathIsArray, wouldn't need to repeat error handling/null checks and also create a reusable function for checking array lengths.



Will spend some more time playing with this later.